### PR TITLE
add ESIGMA generation tutorial and minor bug fixes

### PR DESCRIPTION
**Pull request by Akash-Maurya-0899**
- added ESIGMA generation tutorial Jupyter notebook
    - fixed epochs of inspiral quantities' PyCBC TimeSeries
    such that the last time-grid point falls at 0

### CODE REVIEW

File: gwnr/waveform/esigma_utils.py [hunk 1]
```diff
@@ -302,7 +302,7 @@ def get_inspiral_esigma_modes(
             k: pt.TimeSeries(
                 modes[k].data.data,
                 delta_t=delta_t,
-                epoch=-delta_t * len(modes[k].data.data),
+                epoch=-delta_t * (len(modes[k].data.data)-1),
```
Comment (prayush) [inline gwnr/waveform/esigma_utils.py:305]: Why is this `-1` needed?

File: gwnr/waveform/esigma_utils.py [hunk 2]
```diff
@@ -320,7 +320,7 @@ def get_inspiral_esigma_modes(
         if return_pycbc_timeseries:
             for name in return_orbital_params:
                 exec(
-                    f"orbital_var_dict['{name}'] = pt.TimeSeries({name}.data.data, delta_t=delta_t, epoch=-delta_t * len({name}.data.data)"
+                    f"orbital_var_dict['{name}'] = pt.TimeSeries({name}.data.data, delta_t=delta_t, epoch=-delta_t * (len({name}.data.data)-1))"
```
Comment (prayush) [inline gwnr/waveform/esigma_utils.py:323]: same Q as above

**Review by prayush — APPROVED** [review]
Thanks for adding this, the notebook looks good. I would suggest splitting the commit into 2, 1 where you make changes to `esigma_utils` and 1 where you add the notebook + html.

**Comment by prayush** [comment]
@Akash-Maurya-0899 ping me when you have split the commit into 2 and we can merge this.

File: gwnr/waveform/esigma_utils.py (hunk 1 above)
Comment (Akash-Maurya-0899) [reply gwnr/waveform/esigma_utils.py:305]: Consider a time-grid of $N$ points with grid-spacing $\Delta t$. If the last grid point is to be at $0$ (i.e. $t_{N-1}=0$), then the formula for time at the $i^{\rm{th}}$ grid point ($t_i$) would be,

$t_i = -\Delta t \ (N-1-i); i \in \lbrace 0, 1, 2, ..., N-1 \rbrace$

which gives $t_0 = -\Delta t*(N-1)$.

File: gwnr/waveform/esigma_utils.py (hunk 2 above)
Comment (Akash-Maurya-0899) [reply gwnr/waveform/esigma_utils.py:323]: Same response

**Comment by Akash-Maurya-0899** [comment]
@prayush Split the commit into two.

**Pull request merged**